sirepo/package_data/static/js/sirepo-components.js

```diff
@@ -3007,6 +3007,85 @@ SIREPO.app.directive('commonFooter', function() {
     };
 });
 
+
+SIREPO.app.directive('simConversionModal', function(appState, requestSender) {
+    return {
+        restrict: 'A',
+        scope: {
+            convMethod: '@',
+        },
+        template: `
+            <div data-common-footer="nav"></div>
+            <div data-import-python=""></div>
+            <div data-confirmation-modal="" data-is-required="" data-id="sr-conv-dialog" data-title="Open as a New {{ title }} Simulation" data-modal-closed="resetURL()" data-cancel-text="{{ displayLink() ? \'Close\' : \'Cancel\' }}" data-ok-text="{{ displayLink() ? \'\' : \'Create\' }}" data-ok-clicked="openConvertedSimulation()">
+              <div data-ng-if="! displayLink()"> Create a {{ title }} simulation with an equivalent beamline? </div>
+              <div data-ng-if="displayLink()">
+                {{ title }} simulation created: <a data-ng-click="closeModal()" href="{{ newSimURL }}" target="_blank">{{ newSimURL }} </a>
+              </div>
+            </div>
+        `,
+        controller: function($scope) {
+            $scope.newSimURL = false;
+            $scope.title = $scope.convMethod == 'create_shadow_simulation' ? 'Shadow' : 'SRW';
+
+            function createNewSim(data) {
+                requestSender.sendRequest(
+                    'newSimulation',
+                    simData => {
+                        ['simulationId', 'simulationSerial'].forEach(function(f) {
+                            data.models.simulation[f] = simData.models.simulation[f];
+                        });
+                        data.version = simData.version;
+                        requestSender.sendRequest(
+                            'saveSimulationData',
+                            genSimURL,
+                            data);
+                    },
+                    newSimData(data));
+            }
+
+            function newSimData(data) {
+                const res = appState.clone(data.models.simulation);
+                res.simulationType = data.simulationType;
+                if (! res.name){
+                    res.name = 'newSim';
+                }
+                return res;
+            }
+
+            function genSimURL(data) {
+                $scope.newSimURL = requestSender.formatUrlLocal(
+                    'beamline',
+                    {
```

Comment:
missing trailing comma, but i personally would write it like this: `{ 'simulationId': data.models.simulation.simulationId}`. There only one element.